qdrant_client/common/version_check.py

```diff
@@ -0,0 +1,65 @@
+import warnings
+from typing import Any, Optional
+from collections import namedtuple
+
+import httpx
+
+from qdrant_client.auth import BearerAuth
+
+Version = namedtuple("Version", ["major", "minor", "rest"])
+
+
+def get_server_version(
+    rest_uri: str, rest_headers: dict[str, Any], auth_provider: Optional[BearerAuth]
+) -> Optional[str]:
+    try:
+        response = httpx.get(rest_uri + "/", headers=rest_headers, auth=auth_provider)
+    except Exception as er:
+        warnings.warn(f"Unable to get server version: {er}, default to None")
+        return None
+
+    if response.status_code == 200:
+        version_info = response.json().get("version", None)
+        if not version_info:
+            warnings.warn(f"Unable to parse response from server: {response}, default to None")
+        return version_info
+    else:
+        warnings.warn(f"Unexpected response from server: {response}, default to None")
+    return None
+
+
+def parse_version(version: str) -> Version:
+    if not version:
+        raise ValueError("Version is None")
+    try:
+        major, minor, *rest = version.split(".")
+        return Version(int(major), int(minor), rest)
+    except ValueError as er:
+        raise ValueError(
+            f"Unable to parse version, expected format: x.y.z, found: {version}"
+        ) from er
+
+
+def is_versions_compatible(client_version: Optional[str], server_version: Optional[str]) -> bool:
+    if not client_version:
+        warnings.warn(f"Unable to compare with client version {client_version}")
```

Comment:
I'd rather remove these two warnings, because otherwise the output might be too verbose

```
/.../qdrant_client/qdrant_client/common/version_check.py:18: UserWarning: Unable to get server version: [Errno 61] Connection refused, default to None
  warnings.warn(f"Unable to get server version: {er}, default to None")
/.../qdrant_client/qdrant_client/common/version_check.py:49: UserWarning: Unable to compare with server version None
  warnings.warn(f"Unable to compare with server version {server_version}")
/...qdrant_client/qdrant_client/qdrant_remote.py:205: UserWarning: Qdrant client version 1.10.1 is incompatible with server version None. Major versions should match and minor version difference must not exceed 1. Set check_version=False to skip version check.
  warnings.warn(
```